openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -202,3 +205,23 @@ def partial_update(self, request, username):
             )
 
         return Response(account_settings)
+
+
+class AccountDeactivationView(APIView):
+    """
+    Account deactivation viewset. Currently only supports POST requests.
+    Only admins can deactivate accounts.
+    """
+    permission_classes = (permissions.IsAuthenticated, IsStaffUser)
+
+    def post(self, request, username):
+        """
+        POST /api/user/v1/accounts/{username}/deactivate/
```

Comment:
This seemed to me to be the best way without changing too much of existing logic since only the user can change it's attributes: https://github.com/edx/edx-platform/blob/6730937ef324cba2af38e1eafe6a30ee74129c70/openedx/core/djangoapps/user_api/accounts/views.py#L124-L129
Also ``DELETE`` indicates a resource removal, which this really isn't.